core/src/main/java/com/freiheit/fuava/simplebatch/processor/Processors.java

```diff
@@ -154,17 +150,25 @@
      *
      * This is very similar to
      * {@link #batchFileWriter(String, FileWriterAdapter)}, but for each
-     * persisted file there will be a control file as well.
-     *
-     * @return
+     * persisted file there will be a control file and a log file as well.
      */
-    public static <Input, Output> Processor<Input, Output, BatchProcessorResult<ControlFilePersistenceOutputInfo>> controlledBatchFileWriter(
+    public static <Input, Output> Processor<FetchedItem<Input>, Output, BatchProcessorResult<ControlFilePersistenceOutputInfo>> controlledBatchFileWriter(
             final String dirName,
             final String controlFileEnding,
-            final FileOutputStreamAdapter<List<Input>, List<Output>> adapter
-            ) {
-        return new BatchProcessor<Input, Output, ControlFilePersistenceOutputInfo>( controlledFileWriter( dirName,
-                controlFileEnding, adapter ) );
+            final String logFileEnding,
+            final FileOutputStreamAdapter<List<FetchedItem<Input>>, List<Output>> adapter ) {
+        return Processors.compose(
+                new BatchProcessor<FetchedItem<Input>, Output, ControlFilePersistenceOutputInfo>(
```

Comment:
please add a comment here that informs about the fact, that the BatchProcessor here only processes successful items. Or maybe you can rename it to something like SuccessItemsBatchProcessor ?